### PR TITLE
Document Key Return

### PR DESCRIPTION
**Pull request by tjholm**

### CODE REVIEW

File: pkg/plugins/document/boltdb/boltdb.go [hunk 1]
```diff
@@ -301,8 +301,33 @@ func createDoc(key *sdk.Key) BoltDoc {
...
+	} else {
+		id = doc.Id
+		c = col
+	}
+
 	return &sdk.Document{
 		Content: doc.Value,
+		Key: &sdk.Key{
+			Collection: c,
+			// TODO: need to split out parent key id...
```
Comment (medgar-nitric) [inline pkg/plugins/document/boltdb/boltdb.go:329]: Should remove TODO

File: pkg/plugins/document/dynamodb/dynamodb.go [hunk 2]
```diff
@@ -425,10 +428,41 @@ func marshalQueryResult(
...
+				Name: collection.Name,
+				Parent: &sdk.Key{
+					Collection: &sdk.Collection{
+						Name: collection.Parent.Collection.Name,
+					},
+					Id: pk,
+				},
+			}
+		}
+		// Get the sort key as a string
```
Comment (medgar-nitric) [inline pkg/plugins/document/dynamodb/dynamodb.go:455]: Remove this comment?

**Review by medgar-nitric — DISMISSED** [review]
Looks good, thanks for keeping good test coverage.  Just couple of WIP comments to remove.

File: pkg/plugins/document/boltdb/boltdb.go (hunk 1 above)
Comment (tjholm) [reply pkg/plugins/document/boltdb/boltdb.go:329]: done

File: pkg/plugins/document/dynamodb/dynamodb.go (hunk 2 above)
Comment (tjholm) [reply pkg/plugins/document/dynamodb/dynamodb.go:455]: done

**Review by medgar-nitric — DISMISSED** [review]

**Review by jyecusch — COMMENTED** [review]
Added one PR with a proposed change/fix for boltdb, otherwise looks good and happy to approve.

**Review by jyecusch — APPROVED** [review]

**Review by HomelessDinosaur — APPROVED** [review]

**Pull request merged**